tests/i18n/tests.py

```diff
@@ -1419,10 +1419,23 @@ def test_special_fallback_language(self):
         Some languages may have special fallbacks that don't follow the simple
         'fr-ca' -> 'fr' logic (notably Chinese codes).
         """
-        r = self.rf.get('/')
-        r.COOKIES = {}
-        r.META = {'HTTP_ACCEPT_LANGUAGE': 'zh-my,en'}
-        self.assertEqual(get_language_from_request(r), 'zh-hans')
+        tests = [
+            ('zh-my', 'zh-hans'),
+            # canonical
+            ('zh-Hans-CN', 'zh-hans'),
+            ('zh-Hant-HK', 'zh-hant'),
+            # lowercase variations
+            ('zh-hans-mo', 'zh-hans'),
+            ('zh-hant-tw', 'zh-hant'),
+            ('zh-hans-HK', 'zh-hans'),
+            ('zh-hant-SG', 'zh-hant'),
```

Comment:
These are not related with the `fallback` feature so I moved them to a separate test.